File: 2609-distinct-prime-factors-of-product-of-array/2609-distinct-prime-factors-of-product-of-array.cpp
class Solution {
public:
    int distinctPrimeFactors(vector<int>& nums) {
        set<int>dp;
        for(int i=0;i<nums.size();i++){
            for(int j=2;j*j<=nums[i];j++){
                while(nums[i]%j==0){
                     dp.insert(j);
                     nums[i]/=j;
                }
            }
            if(nums[i]>1){
                dp.insert(nums[i]);
            }    
        }
        return dp.size();
    }
};

File: 2609-distinct-prime-factors-of-product-of-array/2609-distinct-prime-factors-of-product-of-array.py
class Solution:
    def distinctPrimeFactors(self, nums: List[int]) -> int:
        s=set()
        for i in nums:
            for j in range(2,i+1):
                while(i%j==0):
                    s.add(j)
                    i//=j
            
        return len(s)
        
        